Addition of some new indicators and it's unit tests

bi-kash opened the pull request:
@ta-oliver Do you have any suggestions?
1. Addition of new indicators
2. Addition of corresponding tests and reconciliation with ta_library

ta-oliver commented:
@bi-kash - thanks! Architectural question. Do we want to include the main calculation code in InferTrade or could we just import and use the `ta` versions in the allocation rules? We could still include our versions as independent tests with the test suite. We could discuss on a video call?

ta-oliver commented:
@Cal231995 - do you want to try out the new functions? @bi-kash can help you with getting to run the package and new functions on your computer via a video call.

Cal231995 commented:
Yes sir. would really love to try these functions. I'll try to test it for accuracy. Thanks

bi-kash commented:
> @bi-kash - thanks! Architectural question. Do we want to include the main calculation code in InferTrade or could we just import and use the `ta` versions in the allocation rules? We could still include our versions as independent tests with the test suite. We could discuss on a video call?

Using the ta version will be easier compared to the main calculation code. These functions demand for a requirement i.e. Must be a data frame with a close column and all. Using ta will reduce that hassle of course.

ta-oliver commented on infertrade/algos/community/allocations.py:
```diff
@@ -448,36 +432,47 @@ def stochastic_RSI_strategy(df: pd.DataFrame, window: int = 14) -> pd.DataFrame:
-    "level_and_change_regression": {
-        "function": level_and_change_regression,
-        "parameters": {"level_coefficient": 0.1, "change_coefficient": 0.1, "level_and_change_constant": 0.1},
-        "series": ["research"],
```
This is removing the old dictionary entries? We need entries for both.

bi-kash commented:
@ta-oliver @Cal231995 

1. Allocation strategy using indicators directly from ta_library.
2. For now, allocation is 1, -1, or 0 calculated from standard rules. These can be limited to max_alloc and -max_alloc using restrictor allocator.
3. Tests are added.  We are basically repeating the same strategies on tests with signals we implemented. Feels weird to repeat the same thing. 

@ta-oliver Do you have any suggestions?

Cal231995 commented:
> @ta-oliver @Cal231995
> 
> 1. Allocation strategy using indicators directly from ta_library.
> 2. For now, allocation is 1, -1 calculated from standard rules. These can be limited to max_alloc and -max_alloc using restrictor allocator.
> 3. Tests are added.  We are basically repeating the same strategies on tests with signals we implemented. Feels weird to repeat the same thing.
> 
> @ta-oliver Do you suggest moving signals we implemented to the tests folder if it's only going to be used for tests. Do you have any other suggestions?

I will upload a workflow process for the allocation.
Initial idea is to allocate 10% of porfolio for every buy/sell signal

bi-kash commented:
> > @ta-oliver @Cal231995
> > 
> > 1. Allocation strategy using indicators directly from ta_library.
> > 2. For now, allocation is 1, -1 calculated from standard rules. These can be limited to max_alloc and -max_alloc using restrictor allocator.
> > 3. Tests are added.  We are basically repeating the same strategies on tests with signals we implemented. Feels weird to repeat the same thing.
> > 
> > @ta-oliver Do you suggest moving signals we implemented to the tests folder if it's only going to be used for tests. Do you have any other suggestions?
> 
> I will upload a workflow process for the allocation.
> Initial idea is to allocate 10% of porfolio for every buy/sell signal

ta-oliver commented:
>@ta-oliver Do you suggest moving signals we implemented to the tests folder if it's only going to be used for tests? Do you have any other suggestions?

Good question.

I think where we can call an external function, like the ta library, that may be better than adding functional code to the InferTrade package. However an independent test is a strong way to make sure the ta library is behaving as expected. So we could have functions in the signals.py file that are wrappers for the ta functions, then call these signal functions for the allocation rules and then check using the independent implementations in the test files.

ta-oliver commented on infertrade/algos/community/allocations.py:
```diff
@@ -255,16 +257,112 @@ def buy_golden_cross_sell_death_cross(
+    price_above_signal = df["close"] > SMA
+    price_below_signal = df["close"] <= SMA
+
+    df.loc[price_above_signal, PandasEnum.ALLOCATION.value] = 1.0
+    df.loc[price_below_signal, PandasEnum.ALLOCATION.value] = -1.0
```
We could parameterise the maximum investment size as max_investment?

ta-oliver commented:
>Initial idea is to allocate 10% of porfolio for every buy/sell signal

We could supply max_investment as an input parameter, so that 0.1 would be 10% of portfolio.

ta-oliver reviewed (approved):
Excellent work, @bi-kash !
